pkg/zql/processors/README.md

```diff
@@ -0,0 +1,296 @@
+# Processors
+
+A pipeline may contain one or more _processors_ to filter or transform event data. You can imagine the data flowing left-to-right through a processor, with its functionality further determined by arguments you may set. Processor names are case-insensitive.
+
+The following available processors are documented in detail below:
+
+* [`cut`](#cut)
+* [`filter`](#filter)
+* [`head`](#head)
+* [`sort`](#sort)
+* [`tail`](#tail)
+* [`uniq`](#uniq)
+
+**Note**: In the examples below, we'll use the `zq -f table` output format for human readability. Due to the width of the Zeek events used as sample data, you may need to "scroll right" in the output to see some field values.
+
+---
+
+## `cut`
+
+|                           |                                                             |
+| ------------------------- | ----------------------------------------------------------- |
+| **Description**           | Return the data only from the specified named fields.       |
+| **Syntax**                | `cut <field-list>`                                          |
+| **Required<br>arguments** | `<field-list>`<br>One or more comma-separated field names.  |
+| **Optional<br>arguments** | None                                                        |
+| **Caveats**               | The specified field names must exist in the input data. If a non-existent field appears in the `<field-list>`, the returned results will be empty. |
+| **Developer Docs**        | https://godoc.org/github.com/mccanne/zq/proc#Cut            |
+
+#### Example:
+
+To return only the `ts` and `uid` columns of `conn` events:
+
+```
+zq -f table '* | cut ts,uid' conn.log
```

Comment:
I think examples like this will be more readable with the `* |`, is that to happen in a separate pass?

Reply:
@aswan: I assume you meant to say they'd be more readable _without_ the `* |`?

This definitely crossed my mind, and I was squarely on the fence about which way to go. I respect that as soon as a user spends any significant time with `zq`, they'll be very pleased with the shorthand where they can leave off the `* |`, so I understand the benefits of introducing it. OTOH, since these docs may be many users' first exposure to `zq`, I'm nervous of using shorthand right off the bat. At the [top-level ZQL readme](https://github.com/mccanne/zq/blob/docs-init/pkg/zql/README.md) I just got done introducing the concept of how there's separate "search" & "pipeline" portions, so someone who comes right to this page might be confused if they see half of that missing right out of the gate.

I suppose I could repeat what I wrote in the [search all events](https://github.com/mccanne/zq/tree/docs-init/pkg/zql/search-syntax#search-all-events) portion of the Search Syntax page and explain that I'm going to use that same shorthand in many places in that Processors page. My only reason to hesitate on doing that was because the Processors page reads more like a reference: Someone might not linger at the top and instead just click the link to jump ahead to the processor they wanted to learn about, once again missing out on important context.

I'm probably over-thinking it. I'm interested to hear if any other reviewers have an opinion, but if you think the "repeat" proposal I just mentioned would be an improvement, I'd probably be game to change over to that.

Reply:
Whoops, yes I meant "without".
Not saying this should happen here's but an idea for an alternate way to present this: a zql query is a sequence of processors (as opposed to search + processors).  However, best practice is to start with a filter proc, so the shorthand is that you can omit the `filter` proc name on an initial filter.

Reply:
> if any other reviewers have an opinion,

Here's one: it's taken me a bit of rewiring because i'm so used to `* |`, but as I start to switch over, i think it's a nice ux improvement that we should promote. As @philrz and @aswan point out though it does require shifting away from the "search | pipeline".